internal/server/data/identity.go

```diff
@@ -94,24 +95,28 @@ func AssignIdentityToGroups(db *gorm.DB, user *models.Identity, provider *models
 	return nil
 }
 
-func CreateIdentity(db *gorm.DB, identity *models.Identity) error {
+func CreateIdentity(db GormTxn, identity *models.Identity) error {
 	return add(db, identity)
 }
 
-func GetIdentity(db *gorm.DB, selectors ...SelectorFunc) (*models.Identity, error) {
+func GetIdentity(db GormTxn, selectors ...SelectorFunc) (*models.Identity, error) {
 	return get[models.Identity](db, selectors...)
 }
 
-func ListIdentities(db *gorm.DB, p *models.Pagination, selectors ...SelectorFunc) ([]models.Identity, error) {
+func ListIdentities(db GormTxn, p *models.Pagination, selectors ...SelectorFunc) ([]models.Identity, error) {
 	return list[models.Identity](db, p, selectors...)
 }
 
-func DeleteIdentity(db *gorm.DB, id uid.ID) error {
+func DeleteIdentity(db GormTxn, id uid.ID) error {
 	return delete[models.Identity](db, id)
 }
 
-func DeleteIdentities(db *gorm.DB, selectors ...SelectorFunc) error {
-	toDelete, err := ListIdentities(db.Select("id"), nil, selectors...)
+func DeleteIdentities(tx GormTxn, selectors ...SelectorFunc) error {
+	selectID := func(db *gorm.DB) *gorm.DB {
+		return db.Select("id")
+	}
+	selectors = append([]SelectorFunc{selectID}, selectors...)
+	toDelete, err := ListIdentities(tx, nil, selectors...)
```

Comment:
Another place where I had to move a thing to the selectors. I think this is the only case that was not a `Preload`